Test failure in ydb/tests/compatibility/test_vector_index.py.TestVectorIndex.test_vector_index[rolling_stable-25-3-1_to_stable-25-4-1]

ydb/tests/compatibility/test_vector_index.py.TestVectorIndex.test_vector_index[rolling_stable-25-3-1_to_stable-25-4-1]

**Test Failure**

**Build Information:**
- Version: 8ea7ec920e76164ab06fa8c5819d5daeea7508ba
- Build Preset: relwithdebinfo
- Branch: main

**Test Report:** [View report](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/21044181712/ya-main-x86-64/try_1/ya-test.html#FAIL)

**Run Command:**
```
./ya make -ttt --build "relwithdebinfo" -F 'test_vector_index.py::TestVectorIndex::test_vector_index[rolling_stable-25-3-1_to_stable-25-4-1]' ydb/tests/compatibility
```

**Owner:** [TEAM:@ydb-platform/engineering](https://github.com/orgs/ydb-platform/teams/engineering)

**Success rate (last 4 days):**
MAIN: 0% (0/12)

**Test run history:** [link](https://datalens.yandex/34xnbsom67hcq?branch=main&full_name=ydb/tests/compatibility/test_vector_index.py.TestVectorIndex.test_vector_index[rolling_stable-25-3-1_to_stable-25-4-1]&build_type=relwithdebinfo)

More info in [dashboard](https://datalens.yandex/4un3zdm0zcnyr)

**Logs:**
- [log](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/21044181712/ya-main-x86-64/try_1/artifacts/logs/ydb/tests/compatibility/test-results/py3test/testing_out_stuff/test_vector_index.py.TestVectorIndex.test_vector_index.rolling_stable-25-3-1_to_stable-25-4-1.log)
- [logsdir](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/21044181712/ya-main-x86-64/try_1/test_artifacts/ydb-tests-compatibility.zip)

**Error Summary:**
```
ydb/tests/compatibility/test_vector_index.py:243: in test_vector_index
    self.knn_search()
ydb/tests/compatibility/test_vector_index.py:206: in knn_search
    self._do_queries(queries)
ydb/tests/compatibility/test_vector_index.py:186: in _do_queries
    result_sets = session_pool.execute_with_retries(query)
contrib/python/ydb/py3/ydb/query/pool.py:252: in execute_with_retries
    return retry_operation_sync(wrapped_callee, retry_settings)
contrib/python/ydb/py3/ydb/retries.py:137: in retry_operation_sync
    for next_opt in opt_generator:
contrib/python/ydb/py3/ydb/retries.py:98: in retry_operation_impl
    result = YdbRetryOperationFinalResult(callee(*args, **kwargs))
contrib/python/ydb/py3/ydb/query/pool.py:250: in wrapped_callee
    return [result_set for result_set in it]
contrib/python/ydb/py3/ydb/_utilities.py:173: in __next__
    return self._next()
contrib/python/ydb/py3/ydb/_utilities.py:164: in _next
    res = self.wrapper(next(self.it))
contrib/python/ydb/py3/ydb/query/session.py:390: in <lambda>
    lambda resp: base.wrap_execute_query_response(
contrib/python/ydb/py3/ydb/query/base.py:211: in decorator
    return func(rpc_state, response_pb, session_state, *args, **kwargs)
contrib/python/ydb/py3/ydb/query/base.py:229: in wrap_execute_query_response
    issues._process_response(response_pb)
contrib/python/ydb/py3/ydb/issues.py:237: in _process_response
    raise exc_obj(_format_response(response_proto), response_proto.issues)
E   ydb.issues.SchemeError: message: "Type annotation" issue_code: 1030 severity: 1 issues { position { row: 3 column: 29 } message: "At function: KiReadTable!" end_position { row: 3 column: 29 } severity: 1 issues { position { row: 3 column: 29 } message: "Cannot find table \'db.[/Root/Uint8_similarity_inner_product]\' because it does not exist or you do not have access permissions. Please check correctness of table path and user permissions." end_position { row: 3 column: 29 } issue_code: 2003 severity: 1 } } (server_code: 400070)
```

--- a/ydb/tests/compatibility/test_vector_index.py
+++ b/ydb/tests/compatibility/test_vector_index.py
@@ -48,7 +48,7 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
         if prefixed:
             prefix = "user, "
         if overlap:
-            overlap = f"overlap_clusters={overlap}"
+            overlap = f", overlap_clusters={overlap}"
         else:
             overlap = ""
         create_index_sql = f"""
@@ -99,7 +99,7 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
             for vector_type in self.vector_types.keys():
                 for distance in self.targets.keys():
                     for distance_func in self.targets[distance].keys():
-                        table_name = f"{vector_type}_{distance}_{distance_func}{prefixed}"
+                        table_name = f"{vector_type}_{distance}_{distance_func}{prefixed}_"
                         order = "ASC" if distance != "similarity" else "DESC"
                         vector = self.get_vector(f"{vector_type}Vector", 1)
                         where = ""

--- a/ydb/tests/compatibility/test_vector_index.py
+++ b/ydb/tests/compatibility/test_vector_index.py
@@ -15,13 +15,14 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
         self.rows_per_user = 3
         self.index_name = "vector_idx"
         self.vector_dimension = 3
+        # vector type: [ data type, conversion function ]
         self.vector_types = {
-            "Uint8": "Knn::ToBinaryStringUint8",
-            "Int8": "Knn::ToBinaryStringInt8",
-            "Float": "Knn::ToBinaryStringFloat",
+            "Uint8": ["Uint8", "Knn::ToBinaryStringUint8"],
+            "Int8": ["Int8", "Knn::ToBinaryStringInt8"],
+            "Float": ["Float", "Knn::ToBinaryStringFloat"],
         }
         if min(self.versions) >= (26, 1):
-            self.vector_types["Bit"] = "Knn::ToBinaryStringBit"
+            self.vector_types["Bit"] = ["Float", "Knn::ToBinaryStringBit"]
         self.targets = {
             "similarity": {"inner_product": "Knn::InnerProductSimilarity", "cosine": "Knn::CosineSimilarity"},
             "distance": {
@@ -33,8 +34,8 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
 
         yield from self.setup_cluster(extra_feature_flags={"enable_vector_index": True})
 
-    def get_vector(self, type, numb):
-        if type == "FloatVector":
+    def get_vector(self, data_type, numb):
+        if data_type == "Float":
             values = [float(i) for i in range(self.vector_dimension - 1)]
             values.append(float(numb))
             return ",".join(f'{val}f' for val in values)
@@ -78,12 +79,13 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
 
         assert wait_for(predicate, timeout_seconds=100, step_seconds=1), "Error getting index status"
 
-    def _write_data(self, name, vector_type, table_name):
+    def _write_data(self, vector_type, table_name):
+        [data_type, converter] = self.vector_types[vector_type]
         values = []
         for key in range(self.rows_count):
-            vector = self.get_vector(vector_type, key + 1)
+            vector = self.get_vector(data_type, key + 1)
             user = 1 + (key % self.rows_per_user)
-            values.append(f'({key}, {user}, Untag({name}([{vector}]), "{vector_type}Vector"))')
+            values.append(f'({key}, {user}, Untag({converter}([{vector}]), "{vector_type}Vector"))')
 
         sql_upsert = f"""
                 UPSERT INTO `{table_name}` (key, user, vec)
@@ -97,17 +99,18 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
         queries = []
         for prefixed in ['', '_pfx']:
             for vector_type in self.vector_types.keys():
+                [data_type, converter] = self.vector_types[vector_type]
                 for distance in self.targets.keys():
                     for distance_func in self.targets[distance].keys():
                         table_name = f"{vector_type}_{distance}_{distance_func}{prefixed}_"
                         order = "ASC" if distance != "similarity" else "DESC"
-                        vector = self.get_vector(f"{vector_type}Vector", 1)
+                        vector = self.get_vector(data_type, 1)
                         where = ""
                         if prefixed:
                             where = "WHERE user=1"
                         queries.append([
                             True, f"""
-                            $Target = {self.vector_types[vector_type]}(Cast([{vector}] AS List<{vector_type}>));
+                            $Target = {converter}(Cast([{vector}] AS List<{data_type}>));
                             SELECT key, vec, {self.targets[distance][distance_func]}(vec, $Target) as target
                             FROM `{table_name}`
                             {where}
@@ -131,15 +134,16 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
 
     def _get_queries_for(self, prefixed, vector_type, distance, distance_func, overlap):
         queries = []
+        [data_type, converter] = self.vector_types[vector_type]
         table_name = f"{vector_type}_{distance}_{distance_func}{prefixed}_{overlap}"
         order = "ASC" if distance != "similarity" else "DESC"
-        vector = self.get_vector(f"{vector_type}Vector", 1)
+        vector = self.get_vector(data_type, 1)
         where = ""
         if prefixed:
             where = "WHERE user=1"
         queries.append([
             True, f"""
-            $Target = {self.vector_types[vector_type]}(Cast([{vector}] AS List<{vector_type}>));
+            $Target = {converter}(Cast([{vector}] AS List<{data_type}>));
             SELECT key, vec, {self.targets[distance][distance_func]}(vec, $Target) as target
             FROM `{table_name}`
             VIEW `{self.index_name}`
@@ -149,28 +153,28 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
         ])
         # Insert, update, upsert, delete
         key = self.rows_count+1
-        vector = self.get_vector(vector_type, key+1)
+        vector = self.get_vector(data_type, key+1)
         queries.append([
             False, f"""
             INSERT INTO `{table_name}` (key, user, vec)
             VALUES ({key}, {1 + (key) % self.rows_per_user},
-                Untag({self.vector_types[vector_type]}([{vector}]), "{vector_type}Vector"))
+                Untag({converter}([{vector}]), "{vector_type}Vector"))
             """
         ])
-        vector = self.get_vector(vector_type, key+2)
+        vector = self.get_vector(data_type, key+2)
         queries.append([
             False, f"""
             UPDATE `{table_name}` SET user=user+1,
-                vec=Untag({self.vector_types[vector_type]}([{vector}]), "{vector_type}Vector")
+                vec=Untag({converter}([{vector}]), "{vector_type}Vector")
             WHERE key={key}
             """
         ])
-        vector = self.get_vector(vector_type, key+3)
+        vector = self.get_vector(data_type, key+3)
         queries.append([
             False, f"""
             UPSERT INTO `{table_name}` (key, user, vec)
             VALUES ({key}, {1 + (key) % self.rows_per_user},
-                Untag({self.vector_types[vector_type]}([{vector}]), "{vector_type}Vector"))
+                Untag({converter}([{vector}]), "{vector_type}Vector"))
             """
         ])
         queries.append([
@@ -229,7 +233,6 @@ class TestVectorIndex(RollingUpgradeAndDowngradeFixture):
                             table_name = f"{vector_type}_{distance}_{distance_func}{prefixed}_{overlap}"
                             self.create_table(table_name)
                             self._write_data(
-                                name=self.vector_types[vector_type],
                                 vector_type=vector_type,
                                 table_name=table_name,
                             )

